### DISCUSSION TITLE
Allow to add new side panels or override existing ones

### DISCUSSION THREAD

**Discussion by SebCorbin** [wagtail/wagtail#10799] [Ideas]
I find myself for the second time needing a way to override or add side_panels in the UI:
1. Added a `BasePreviewSidePanel`-based side panel for a `BaseGenericSetting` so that the input settings (in my case site colors) are previewable. For that I added a custom context_processor (yes, a global one) acting on the url condition to inject the `side_panels` into the context
2. Wanted to add a custom side_panel for a Page, but there's no way to augment this

Should we add a hook for this?

**Comment by SebCorbin** [comment]
Can probably be done once https://github.com/wagtail/wagtail/pull/6025 is in, as we can override context data

**Comment by laymonage** [comment]
For previewing settings, I'd rather have https://github.com/wagtail/wagtail/issues/11347#issuecomment-2122159460 implemented rather than adding a hook.

**Comment by SebCorbin** [comment]
Closing as it's now a feature request https://github.com/wagtail/wagtail/issues/12179 and work must be done in https://github.com/wagtail/wagtail/issues/12463 before
